packages/shared/lib/services/proxy.service.unit.test.ts

```diff
@@ -505,6 +505,97 @@ describe('Proxy service Construct URL Tests', () => {
+                                meta: {},
+                                type: 'ratelimited',
+                                userPresentableMessage: 'Rate limit exceeded. Only 1200 requests are allowed per 1 hour.'
+                            },
+                            message:
+                                'Rate limit exceeded. Only 1200 requests are allowed per 1 hour. For more information see our developer docs at: https://developers.linear.app/docs/graphql/working-with-the-graphql-api/rate-limiting'
+                        }
+                    ]
+                },
+                statusText: 'Bad Request',
+                config: {} as InternalAxiosRequestConfig
+            } as AxiosResponse
+        } as AxiosError;
+        const config = {
+            provider: {
+                auth_mode: 'OAUTH2',
+                proxy: {
+                    retry: {
+                        at: 'x-ratelimit-requests-reset'
+                    }
+                }
+            },
+            token: 'some-oauth-access-token'
+        } as ApplicationConstructedProxyConfiguration;
+        const before = Date.now();
+        const willRetry = await proxyService.retry(config, [], mockAxiosError, 0);
+        const after = Date.now();
+        const diff = after - before;
+        expect(diff).toBeGreaterThan(1000);
+        expect(diff).toBeLessThan(2000);
+        expect(willRetry).toBe(true);
+    });
+
+    it('Should not retry based on the header if it does not match', async () => {
+        const nowInSecs = Date.now() / 1000;
+        const mockAxiosError = {
+            response: {
+                status: 400,
+                code: 400,
+                headers: {
+                    'x-ratelimit-requests-reset': nowInSecs + 1
+                },
+                data: {
+                    errors: [
+                        {
+                            extensions: {
+                                userError: true,
+                                code: 'RATELIMITED',
+                                meta: {},
+                                type: 'ratelimited',
+                                userPresentableMessage: 'Rate limit exceeded. Only 1200 requests are allowed per 1 hour.'
+                            },
+                            message:
+                                'Rate limit exceeded. Only 1200 requests are allowed per 1 hour. For more information see our developer docs at: https://developers.linear.app/docs/graphql/working-with-the-graphql-api/rate-limiting'
+                        }
+                    ]
+                },
+                statusText: 'Bad Request',
+                config: {} as InternalAxiosRequestConfig
+            } as AxiosResponse
+        } as AxiosError;
+        const config = {
+            provider: {
+                auth_mode: 'OAUTH2',
+                proxy: {
+                    retry: {
+                        at: 'x-ratelimit-requests-resets'
```

Comment:
I would make it very different from the real header name. It took me a few seconds to spot the extra `s`

Reply:
Updated